tests/test_enterprise/views/test_grant_data_sharing_permissions.py

```diff
@@ -594,13 +596,10 @@ def test_post_course_specific_consent(
 
         assert resp.url.endswith(expected_redirect_url)  # pylint: disable=no-member
         assert resp.status_code == 302
-        if not defer_creation:
-            dsc.refresh_from_db()
-            assert dsc.granted is consent_provided
 
-        # we'll only create an enrollment record if (1) creation is not deferred, (2) consent has not been provided,
+        # we'll only create an enrollment record if (1) creation is not deferred, (2) consent has been provided,
```

Comment:
Given our confusion between "consent was previously provided" and "consent is currently provided in the post request", it seems worth making the distinction in this comment

Reply:
Reasonable, that comment was also just wrong about when we create the enrollment record. But I'll add a distinction for "when the user is giving consent without having previously consented"

Reply:
The comment was referring to how consent shouldn't have previously been provided, bringing us full circle to the initial point of confusion

Reply:
Well the original comment and the if statement in the test were both just wrong about the conditions for when the enrollment record got created in the view which is what I mean about the comment being wrong.